### PR TITLE
Change Assignments per Registration

### PR DESCRIPTION
**Pull request by erikerikson**
Using a stage-specific, falling back to default, configuration variable from the private.yml

### CODE REVIEW

File: product-photos/0.processor/processor.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ const constants = {
   // methods
   METHOD_START_EXECUTION: 'startExecution',
   // values
+  ASSIGNMENTS_PER_REGISTRATION: process.env.STAGE === 'demo' ? 1 : 3,
```
Comment (laurwang) [inline product-photos/0.processor/processor.js]: Something like 
ASSIGNMENTS_PER_REGISTRATION: process.env.ASSIGNMENTS_PER_REGISTRATION ? process.env.ASSIGNMENTS_PER_REGISTRATION : 1

**Comment by gwsii** [comment]
Perhaps the right thing to do in order to control the behavior of the kinesis-handler code would be to quickly add a "feature flag" environment variable. One that is only set when you want the above "hack" in place? Or perhaps to effectively control the order guarantee (sync or async message handling)?

**Comment by laurwang** [comment]
Kinesis-handler already does asynch vs synch handling, and that's not really part of the problem here, though I guess it means I'd only have to fix it for the async handler.  Anyhow, I'd rather not knock one hack out with another, so I'll soend a few hours trying to come up with a slightly slicker change to kinesis-handler that will allow the photo-processor to basically convert from ingress to egress schema, which seems to be what's really going on here.

**Comment by gwsii** [comment]
Given that @laurwang has provided another mechanism, is this PR still applicable?

File: product-photos/0.processor/processor.js (hunk 1 above)
Comment (erikerikson) [inline product-photos/0.processor/processor.js]: I think that's exactly right.
I'd like the serverless.yml to have a 'behaviors' attribute under which things like this are specified.  The value seems like it should be called 'assignmentsPerRegistration'.  Lastly, to make it stage specific, I'd suggest specifying the value using the stage with a fallback to the non-state specific value...

```
behaviors:
  assignmentsPerRegistration:
    default: 3
    demo: 1
    austin: 1
```

That would make the serverless.yml contents:
```
functions:
  processor:
    environment:
      ASSIGNMENTS_PER_REGISTRATION: ${self:custom.private.behaviors.assignmentsPerRegistration.${opt:stage}, self:custom.private.behaviors.assignmentsPerRegistration.default}
```

Then your code is perfect.

**Comment by erikerikson** [comment]
Hi @gwsii, there are two concerns here.  The configurability change that this PR is about and the septate need to augment the event with the eventID provided by Kinesis.  Pretty sure the original purpose of the PR (to allow assignments power registration to be configurable) is still valid unless I'm missing something...

File: product-photos/0.processor/processor.js (hunk 1 above)
Comment (erikerikson) [inline product-photos/0.processor/processor.js]: slight mod after dev.  the environment variable will fail without a value so there's no need to check the validity of `process.env.ASSIGNMENTS_PER_REGISTRATION`.  commit incoming...

**Comment by erikerikson** [comment]
Thanks @laurwang for un-hacking this!  Should be ready for actual merge now but please take a look and test it out a bit.

**Pull request merged**